src/dynamicprompts/samplers/combinatorial.py

```diff
@@ -32,136 +30,113 @@ def _dedupe(arr: list[str]) -> tuple[str, ...]:
 
 
 def _combo_to_prompt(
-    sampler_router: SamplerRouter,
+    sampling_context: SamplingContext,
     combo: list[Command],
 ) -> typing.Iterable[list[str]]:
     if len(combo) == 0:
         yield []
         return
 
     c_1, c_rest = combo[0], combo[1:]
-    gen = sampler_router.generator_from_command(c_1)
-    if c_1.sampling_method != SamplingMethod.COMBINATORIAL:
-        for rest_prompt in _combo_to_prompt(sampler_router, c_rest):
+    gen = sampling_context.generator_from_command(c_1)
+    if sampling_context.get_effective_sampling_method(c_1).is_nonfinite():
+        for rest_prompt in _combo_to_prompt(sampling_context, c_rest):
             val = next(gen)
             yield [val] + rest_prompt
     else:
         for p in gen:
-            for rest_prompt in _combo_to_prompt(sampler_router, c_rest):
+            for rest_prompt in _combo_to_prompt(sampling_context, c_rest):
                 if rest_prompt:
                     yield [p] + rest_prompt
                 else:
                     yield [p]
```

Comment:
There is a lot going on in this code. Do you have any ideas on how we can simplify it? The first part of the `if `statement says that we have a command that doesn't behave like a combinatorial command, and therefore we need to sample it out of band. The `else` handles the sampling combinatorially. 

Some thoughts:

a) Swap the order of the statements. The non-finite section is the exceptional condition so would make sense to be read second
b) wrap lines 43-45 in a function called something like `def sample_with_nonfinite_command(nonfinite_command, other_commands) -> Iterable[list_of_tokens]`
c) wrap lines 47-52 in a function called something like `def sample_with_finite_command(finite_command, other_command) -> Iterable[list_of_tokens]`

What do you think?